packages/next/shared/lib/dynamic.tsx

```diff
@@ -110,22 +112,46 @@ export default function dynamic<P = {}>(
   // Support for passing options, eg: dynamic(import('../hello-world'), {loading: () => <p>Loading something</p>})
   loadableOptions = { ...loadableOptions, ...options }
 
+  if (!process.env.__NEXT_REACT_ROOT) {
```

Comment:
I think we want this to be the `concurrentFeatures` flag instead, for consistency.